docs/arch/adr-25.md

```diff
@@ -0,0 +1,157 @@
+{"annotations": [
+  {
+    "task": "T0",
+    "value": [
+      {
+        "frame": 0,
+        "tool": 0,
+        "x": 452.18341064453125,
+        "y": 202.87478637695312,
+        "details": [
+          {"value": 0},
+          {"value": [
+            {"option-1": 1},
+            {"option-2": 1},
+            {"None": 1}
+          ]}
+        ]
+      },
+      {
+        "frame": 0,
+        "tool": 1,
+        "x": 404.61279296875,
+        "y": 583.4398803710938,
+        "details": [
+          {"value": 1},
+          {"value": [
+          {"option-3": 1},
+          {"option-4": 1},
+          {"option-5": 1}
+        ]}
+      ]
+    }
+  ]}
+]}
+```
+
+This annotation structure has a few issues:
+
+- the annotations in the `details` array have no indication of the task types these originally are
+- the annotations in the `details` array does not have an unique identifier
+
+The missing task types and unique identifiers make it difficult to make downstream analysis and aggregation scripts. The aggregation code now has to parse the details list and make a "mock annotation" of the correct structure to be passed along to the next reducer.
+
+### Sub-task UI
+
+The sub-task UI positioned itself fixed below relative to the position of the mark. Notably transcription projects have commented that this interferes with being able to transcribe successfully since the dialog may cover up part of the subject and cannot be moved without moving the drawing mark.
+
+## Decision
+
+For initial support, we will support the single and multiple choice question tasks and the text task in the sub-task. Slider task may be deprecated and dropdown task may be changing in ways we do not have a plan for yet, so they can be supported later if it makes sense to add them.
+
+### Annotation JSON structure
+
+The annotations in the details array will be updated to be an object that just contains a reference to the task's unique identifier. The task annotation itself will be stored in the classification's annotations flattened. 
+
+The main benefit of this reorganization will be with downstream analysis and aggregation. When aggregating drawn shapes the first step is clustering. Once the clusters are found the subtasks need to be aggregated within each cluster. This will be easier to do if the structure of each subtask annotation is the same as if that task was asked on its own. The code can just take all subtask annotations within a cluster and just pass it to the reducer as if it is a list of main task annotations without having to reshape them.
+
+An example of the new sub-task annotation JSON structure:
+
+```json
+{"annotations": [
+  {
+    "task": "T0",
+    "taskType": "drawing",
+    "value": [
+      {
+        "frame": 0,
+        "tool": 0,
+        "toolType": "point",
+        "x": 452.18341064453125,
+        "y": 202.87478637695312,
+        "details": [
+          {"task": "T0.0.0"},
```

Comment:
I've added a comment in the consequences section about potential benefits to the flatter structure outside of aggregation. I'd like more researcher input into this specific suggestion though.

Reply:
It occurred to me that if we gave annotations identifiers, then `details` here could be an array of annotation IDs, each pointing to an annotation in `classification.annotations`.

Reply:
I think we've discussed in the past that `details` wasn't a very good choice of name for the mark's task annotations array. We could change it to `annotations` for clarity.

Reply:
@CKrawczyk or @lcjohnso do you have a preference? I'm not sure if `annotations` is clear enough to differentiate from the parent `annotations` key in the classification... but I agree `details` isn't very good either.

Reply:
The code uses `annotations` because both `classification` and `mark` use the `AnnotationStore` model. That may have been premature optimisation: the idea being that the subtask popup box should act exactly like the task area, only updating `mark` instead of `classification`.

Reply:
As it is a sub-task would `sub_annotations` (or similar) work?

Reply:
Off the top of my head, I think that depends on whether the task components can be updated to take either `annotations` or `sub_annotations` as a prop.

Reply:
We're conflating in this discussion what the internal stores and components are doing with what is being submitted in the classification annotation. The Task components can take whatever we want them to take as a prop. The question I'm posing has to do with the naming at classification submission.

Reply:
#1432 allows us to pass any annotation into a Task component, regardless of where it comes from or what the source is called. At the moment, master is hard-coded to use `classification.annotations`. That's my fault.

Reply:
There isn't a clear resolution in this discussion on what to do, but we have issue #1443 now to track specifically the transformations needing to happen before submitting the annotations as a classification, so any further discussion can happen there and we can update the ADR as needed.